src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -181,6 +181,7 @@ public override Binder VisitMethodDeclaration(MethodDeclarationSyntax methodDecl
                     if (usage == NodeUsage.MethodBody)
                     {
                         method = method ?? GetMethodSymbol(methodDecl, resultBinder);
+                        Debug.Assert(method is not null);
```

Comment:
@dotnet/roslyn-compiler wanted some feedback here. The two `Debug.Assert` calls are seemingly very likely identifying the cause for the NRT in #73022. Pretty much every other call in this type to `GetMethodSymbol` handles `null` as a valid return value and simply avoids creating the `InMethodBinder`. I'm leaning towards doing the same here but at the same time I can't find a way to construct a test case that hits the `null` so I'm reluctant to do that.